odoo/orm/fields_textual.py

```diff
@@ -280,8 +288,66 @@ def write(self, records, value):
         # Maybe we can use Cache.update(records.with_context(cache_update_raw=True), self, new_translations_list, dirty=True)
         cache.update_raw(records, self, new_translations_list, dirty=True)
 
+    def to_sql(self, model: BaseModel, alias: str, flush: bool = True) -> SQL:
+        sql_field = super().to_sql(model, alias, flush)
+        if self.translate:
+            langs = self.get_translation_fallback_langs(model.env)
+            sql_field_langs = [SQL("%s->>%s", sql_field, lang) for lang in langs]
+            if len(sql_field_langs) == 1:
+                return sql_field_langs[0]
+            return SQL("COALESCE(%s)", SQL(", ").join(sql_field_langs))
+        return sql_field
+
+    def to_raw_sql(self, model: BaseModel, alias: str, flush: bool = True) -> SQL:
```

Comment:
Because the logic of the field is leaked in `models.py` which needs to get the sql for the field without the translation to the language of the context. Once that logic is moved inside this file, the function can disappear.

Reply:
Oh damned, I just got it!  As it calls `super().to_sql()`, it is actually equivalent to `self.to_sql()` but _without the override in the current class_!  It is totally non-standard OO programming, and a hack that is only possible in Python.  Why didn't you simply call the parent class (`Field`) explicitly?  This would have been less confusing.  This requires _at least_ a comment.

Reply:
Comment added.